module/gui/Tools.py

```diff
@@ -0,0 +1,525 @@
+import logging
+
+def whatsThisFormat(title, text):
+    #html = "<b>" + title + "</b><br><br>" + text
+    html = "<table><tr><td><b>" + title + "</b></td></tr></table>" + "<table><tr><td>" + text + "</td></tr></table>"
+    return html
+
+class WhatsThisButton(QPushButton):
+    def __init__(self, text="?"):
+        QPushButton.__init__(self, text)
+        self.text = text
+        self.setDefault(False)
+        self.setAutoDefault(False)
+        self.setFocusPolicy(Qt.NoFocus)
+        self.updateSize()
+        self.connect(self, SIGNAL("clicked()"), QWhatsThis.enterWhatsThisMode)
+
+    def updateSize(self):
+        width  = self.fontMetrics().boundingRect(self.text).width()
+        height = self.fontMetrics().boundingRect(self.text).height()
+        s = width
+        if height > s:
+            s = height
+        s += 4  # border
+        self.setFixedSize(s, s)
+
+class WtDialogButtonBox(QDialogButtonBox):
+    def __init__(self, orientation=Qt.Horizontal, parent=None):
+        QDialogButtonBox.__init__(self, orientation, parent)
+        self.wtBtn = WhatsThisButton()
+        self.l = QHBoxLayout()
+        self.l.setContentsMargins(0, 0, 0, 0)
+        self.l.setSpacing(0)
+        self.l.addWidget(self.wtBtn)
+        self.spacer = QSpacerItem(self.wtBtn.width(), self.wtBtn.width())
+        self.l.addSpacerItem(self.spacer)
+        self.l.addWidget(self)
+
+    def layout(self):
+        return self.l
+
+    def hideWhatsThisButton(self, hide=True):
+        self.wtBtn.setHidden(hide)
+        if hide:
+            self.spacer.changeSize(0, 0)
+        else:
+            self.spacer.changeSize(self.wtBtn.width(), self.wtBtn.width())
+
+    def updateWhatsThisButton(self):
+        if not self.wtBtn.isHidden():
+            self.wtBtn.updateSize()
+            self.spacer.changeSize(self.wtBtn.width(), self.wtBtn.width())
+
+class LineView(QLineEdit):
+    def __init__(self, contents=""):
+        QLineEdit.__init__(self, contents)
+        # read-only
+        self.setReadOnly(True)
+        # set background color
+        self.pal = QPalette(self.palette())
+        newcolor = QColor(QLabel().palette().color(QPalette.Active, QLabel().backgroundRole()))
+        self.pal.setColor(QPalette.Active, self.backgroundRole(), newcolor)
+        self.pal.setColor(QPalette.Inactive, self.backgroundRole(), newcolor)
+        self.setPalette(self.pal)
+        # remove from taborder
+        self.setFocusPolicy(Qt.NoFocus)
+        # prohibit text selection
+        self.connect(self, SIGNAL("selectionChanged()"), self.deselect)
+        # disable context menu
+        self.setContextMenuPolicy(Qt.NoContextMenu)
+
+class MessageBox(QDialog):
+    """
+        icon:               btnSet:
+          N NoIcon            OK
+          Q Question          OK_CANCEL
+          I Information       YES_NO
+          W Warning
+          C Critical
+    """
+
+    def __init__(self, parent, text, icon, btnSet, noTranslation=False):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [MessageBox.__init__ is too complex (19)](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)